mlir/lib/Dialect/SCF/Transforms/ParallelForToNestedFors.cpp

```diff
@@ -0,0 +1,80 @@
+//===- ParallelForToNestedFors.cpp - scf.parallel to nested scf.for ops --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Transforms SCF.ParallelOp to nested scf.for ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Passes.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/IR/PatternMatch.h"
+#include "llvm/Support/Debug.h"
+
+namespace mlir {
+#define GEN_PASS_DEF_SCFPARALLELFORTONESTEDFORS
+#include "mlir/Dialect/SCF/Transforms/Passes.h.inc"
+} // namespace mlir
+
+using namespace mlir;
+
+FailureOr<scf::LoopNest>
+mlir::scf::parallelForToNestedFors(RewriterBase &rewriter,
+                                   scf::ParallelOp parallelOp) {
+
+  if (!parallelOp.getResults().empty()) {
+    parallelOp->emitError("Currently scf.parallel to scf.for conversion "
+                          "doesn't support scf.parallel with results.");
+    return failure();
```

Comment:
I think there is no need to emit an error here. Better to use rewriter.notifyMatchFailure IMO.
```suggestion
    return rewriter.notifyMatchFailure(
      parallelOp, "Currently scf.parallel to scf.for conversion "
                  "doesn't support scf.parallel with results.");
```